packages/core-services-mongodb-remote/src/lib/internal/CoreRemoteMongoCollectionImpl.ts

```diff
@@ -0,0 +1,287 @@
+import { ObjectID } from "bson";
+import { Codec, CoreStitchServiceClient } from "stitch-core";
+import RemoteCountOptions from "../RemoteCountOptions";
+import RemoteDeleteResult from "../RemoteDeleteResult";
+import RemoteFindOptions from "../RemoteFindOptions";
+import RemoteInsertManyResult from "../RemoteInsertManyResult";
+import RemoteInsertOneResult from "../RemoteInsertOneResult";
+import RemoteUpdateOptions from "../RemoteUpdateOptions";
+import RemoteUpdateResult from "../RemoteUpdateResult";
+import CoreRemoteMongoCollection from "./CoreRemoteMongoCollection";
+import CoreRemoteMongoReadOperation from "./CoreRemoteMongoReadOperation";
+import ResultDecoders from "./ResultDecoders";
+
+export default class CoreRemoteMongoCollectionImpl<T>
```

Comment:
[nit] `query` -> `filter` everywhere in this file, except when passing along as an arg to the Stitch server